src/main/java/org/openstreetmap/atlas/geography/atlas/builder/proto/ProtoAtlasBuilder.java

```diff
@@ -0,0 +1,237 @@
+        return protoLocationBuilder.build();
+    }
+
+    private Location convertProtoLocationToLocation(final ProtoLocation protoLocation)
+    {
+        final Longitude longitude = Longitude.dm7(protoLocation.getLongitude());
+        final Latitude latitude = Latitude.dm7(protoLocation.getLatitude());
+        return new Location(latitude, longitude);
+    }
+
+    private void parseLines(final PackedAtlasBuilder builder, final List<ProtoLine> lines)
+    {
+        lines.forEach(line ->
+        {
+            final long identifier = line.getId();
+            final List<Location> shapePoints = line.getShapePointsList().stream()
+                    .map(this::convertProtoLocationToLocation).collect(Collectors.toList());
+            final PolyLine geometry = new PolyLine(shapePoints);
+            final Map<String, String> tags = parseTags(line.getTagsList());
+            builder.addLine(identifier, geometry, tags);
+        });
+    }
+
+    private void parsePoints(final PackedAtlasBuilder builder, final List<ProtoPoint> points)
+    {
+        points.forEach(point ->
+        {
+            final long identifier = point.getId();
+            final Longitude longitude = Longitude.dm7(point.getLocation().getLongitude());
+            final Latitude latitude = Latitude.dm7(point.getLocation().getLatitude());
+            final Location geometry = new Location(latitude, longitude);
+            final Map<String, String> tags = parseTags(point.getTagsList());
+            builder.addPoint(identifier, geometry, tags);
+        });
+    }
+
+    private Map<String, String> parseTags(final List<ProtoTag> tagsList)
+    {
+        try
+        {
+            final Map<String, String> result = Maps.hashMap();
+            for (final ProtoTag tag : tagsList)
+            {
+                final StringList tagSplit = StringList.split(tag.getTagText(), TAG_DELIMITER);
+                if (tagSplit.size() == 2)
+                {
+                    result.put(tagSplit.get(0), tagSplit.get(1));
+                }
+                if (tagSplit.size() == 1)
+                {
+                    result.put(tagSplit.get(0), "");
+                }
+            }
+            return result;
+        }
+        // TODO do something smarter here?
+        catch (final Throwable error)
+        {
+            throw new CoreException("Unable to parse tags.");
+        }
+    }
+
+    private void writeLinesToBuilder(final Atlas atlas,
+            final ProtoAtlasContainer.Builder protoAtlasBuilder)
+    {
+        long numberOfLines = 0;
+
+        for (final Line line : atlas.lines())
+        {
+            final ProtoLine.Builder protoLineBuilder = ProtoLine.newBuilder();
+            protoLineBuilder.setId(line.getIdentifier());
+
+            final List<ProtoLocation> protoLocations = StreamSupport
+                    .stream(line.getRawGeometry().spliterator(), false)
```

Comment:
An alternative way (not sure if it's cleaner) would be `line.asPolyLine().stream().map(...).collect(...);`

Reply:
I'll change this. For some reason I couldn't get that to work initially so went with the StreamSupport solution. I was probably just spelling something wrong.